samples/android/tutorial-4-opencl/jni/GLrender.cpp

```diff
@@ -0,0 +1,315 @@
+#include <GLES2/gl2.h>
+#include <GLES2/gl2ext.h>
+
+#include "common.hpp"
+
+float vertexes[] = {
+        -1.0f, -1.0f,
+        -1.0f,  1.0f,
+         1.0f, -1.0f,
+         1.0f,  1.0f
+};
+float texCoordOES[] = {
+        0.0f, 1.0f,
+        0.0f, 0.0f,
+        1.0f, 1.0f,
+        1.0f, 0.0f
+};
+float texCoord2D[] = {
+        0.0f, 0.0f,
+        0.0f, 1.0f,
+        1.0f, 0.0f,
+        1.0f, 1.0f
+};
+
+const char vss[] = \
+            "attribute vec2 vPosition;\n" \
+            "attribute vec2 vTexCoord;\n" \
+            "varying vec2 texCoord;\n" \
+            "void main() {\n" \
+            "  texCoord = vTexCoord;\n" \
+            "  gl_Position = vec4 ( vPosition.x, vPosition.y, 0.0, 1.0 );\n" \
+            "}";
+
+const char fssOES[] = \
+            "#extension GL_OES_EGL_image_external : require\n" \
+            "precision mediump float;\n" \
+            "uniform samplerExternalOES sTexture;\n" \
+            "varying vec2 texCoord;\n" \
+            "void main() {\n" \
+            "  gl_FragColor = texture2D(sTexture,texCoord);\n" \
+            "}";
+
+const char fss2D[] = \
+            "precision mediump float;\n" \
+            "uniform sampler2D sTexture;\n" \
+            "varying vec2 texCoord;\n" \
+            "void main() {\n" \
+            "  gl_FragColor = texture2D(sTexture,texCoord);\n" \
+            "}";
+
+int progOES = 0;
```

Comment:
it's better to use `GLuint` for programs too (see `glCreateProgram` description)